cfme/tests/cloud_infra_common/test_html5_vm_console.py

```diff
@@ -0,0 +1,168 @@
+from cfme.configure import configuration
+from cfme.common.vm import VM
+from utils import testgen, version, ssh
+from utils.appliance.implementations.ui import navigate_to
+from utils.log import logger
+from utils.conf import credentials
+from wait_for import wait_for
+
+pytestmark = pytest.mark.usefixtures('setup_provider')
+
+pytest_generate_tests = testgen.generate(
+    [OpenStackProvider, RHEVMProvider, VMwareProvider],
+    scope='module'
+)
+
+
+@pytest.fixture(scope="function")
+def vm_obj(request, provider, setup_provider, small_template, vm_name):
+    """
+    Create a VM on the provider with the given template, and return the vm_obj.
+
+    Also, it will remove VM from provider using nested function _delete_vm
+    after the test is completed.
+
+    """
+    vm_obj = VM.factory(vm_name, provider, template_name=small_template)
+
+    @request.addfinalizer
+    def _delete_vm():
+        try:
+            vm_obj.delete_from_provider()
+        except Exception:
+            logger.warning("Failed to delete vm `{}`.".format(vm_obj.name))
+
+    vm_obj.create_on_provider(timeout=2400, find_in_cfme=True, allow_skip="default")
+    return vm_obj
+
+
+def _configure_vmware_console_for_test(appliance):
+    """Configure VMware Console to use VNC which is what is required for the HTML5 console."""
+    navigate_to(appliance.server, 'Server')
+
+    settings_pg = VMwareConsoleSupport(
+        appliance=appliance,
+        console_type='VNC',
+    )
+    settings_pg.update()
+
+
+def _configure_server_roles_for_test():
+    """
+    Enable websocket role if it is disabled.
+
+    Currently the fixture cfme/fixtures/base.py,
+    disables the websocket role to avoid intrusive popups.
+    """
+    # TODO:Find correct place to put this method at.
+    roles = configuration.get_server_roles()
+    if 'websocket' in roles and not roles['websocket']:
+        logger.info('Enabling the websocket role to allow console connections')
+        roles['websocket'] = True
+        configuration.set_server_roles(**roles)
+
+
+@pytest.mark.uncollectif(lambda: version.current_version() < '5.8', reason='Only valid for >= 5.8')
+def test_html5_vm_console(appliance, provider, vm_obj):
+    """
+    Test the HTML5 console support for a particular provider.
+
+    The supported providers are:
+
+        VMware
+        Openstack
+        RHV
+
+    For a given provider, and a given VM, the console will be opened, and then:
+
+        - The console's status will be checked.
+        - A command that creates a file will be sent through the console.
+        - Using ssh we will check that the command worked (i.e. that the file
+          was created.
+    """
+    # temporarily getting credentials from vspherehost-nested because they
+    # match with the PhotonOS Credentials, but need to add credentials for
+    # templates in the credentials.eyaml
+    photon_os_username = credentials['vspherehost-nested'].get('username')
```

Comment:
These should be replaced with template credentials, but since we do not have template credentials in our yamls, I have used these credentials of "vspherehost-nested" here, but I will put up a separate GitLab PR to add Template credentials in our yamls and then those will be referenced here.